go/vt/schemadiff/errors.go

```diff
@@ -334,3 +334,39 @@ type ViewDependencyUnresolvedError struct {
 func (e *ViewDependencyUnresolvedError) Error() string {
 	return fmt.Sprintf("view %s has unresolved/loop dependencies", sqlescape.EscapeID(e.View))
 }
+
+type InvalidColumnReferencedInViewError struct {
+	View      string
+	Table     string
+	Column    string
+	Ambiguous bool
+}
+
+func (e *InvalidColumnReferencedInViewError) Error() string {
+	switch {
+	case e.Column == "":
+		return fmt.Sprintf("view %s references non-existent table %s", sqlescape.EscapeID(e.View), sqlescape.EscapeID(e.Table))
+	case e.Table != "":
+		return fmt.Sprintf("view %s references non-existent column %s.%s", sqlescape.EscapeID(e.View), sqlescape.EscapeID(e.Table), sqlescape.EscapeID(e.Column))
+	case e.Ambiguous:
+		return fmt.Sprintf("view %s references unqualified but non unique column %s", sqlescape.EscapeID(e.View), sqlescape.EscapeID(e.Column))
+	default:
+		return fmt.Sprintf("view %s references unqualified but non-existent column %s", sqlescape.EscapeID(e.View), sqlescape.EscapeID(e.Column))
```

Comment:
the error message looks very fragile based on what struct fields are set.

Reply:
can we also create the message ahead of time, when the struct is created?

Reply:
I think that’s actually the wrong pattern. The core is the structured error and the string format is just a detail here. 

The structured error removes the issue of string being brittle because anyone wants more details doesn’t look at the string at all.

Reply:
> the error message looks very fragile based on what struct fields are set.

I can rewrite that to be more clear

Reply:
I was able to simplify `InvalidColumnReferencedInViewError` by removing unused `Table` columns and subsequent error message generation cases.

Reply:
@harshit-gangal allocating the string is also wasteful since when using structured errors it is never used. So I don’t think we should do that.

Reply:
At the MySQL protocol level, error strings are transferred over the wire, so we might be generating them eventually.

Reply:
@harshit-gangal this is never used for MySQL level errors. But even then, we’d allocate when needed and not always so then it’s imho still an improvement. 

It also improved the ergonomics imho since it’s now easier to create a structured error then.